test/test_pebble.py

```diff
@@ -0,0 +1,512 @@
+            "status": "OK",
+            "status-code": 200,
+            "type": "sync"
+        })
+        change = self.api.get_change('70')
+        self.assert_mock_change(change)
+        self.assertEqual(self.api.requests, [
+            ('GET', '/v1/changes/70', None, None),
+        ])
+
+    def test_abort_change(self):
+        self.api.responses.append({
+            "result": self.get_mock_change_dict(),
+            "status": "OK",
+            "status-code": 200,
+            "type": "sync"
+        })
+        change = self.api.abort_change('70')
+        self.assert_mock_change(change)
+        self.assertEqual(self.api.requests, [
+            ('POST', '/v1/changes/70', None, {'action': 'abort'}),
+        ])
+
+    def _test_services_action(self, action, api_func, services):
+        self.api.responses.append({
+            "change": "70",
+            "result": None,
+            "status": "Accepted",
+            "status-code": 202,
+            "type": "async"
+        })
+        change = self.get_mock_change_dict()
+        change['ready'] = False
+        self.api.responses.append({
+            "result": change,
+            "status": "OK",
+            "status-code": 200,
+            "type": "sync"
+        })
+        change = self.get_mock_change_dict()
+        change['ready'] = True
+        self.api.responses.append({
+            "result": change,
+            "status": "OK",
+            "status-code": 200,
+            "type": "sync"
+        })
+        change_id = api_func()
+        self.assertEqual(change_id, '70')
+        self.assertEqual(self.api.requests, [
+            ('POST', '/v1/services', None, {'action': action, 'services': services}),
+            ('GET', '/v1/changes/70', None, None),
+            ('GET', '/v1/changes/70', None, None),
+        ])
+
+    def _test_services_action_async(self, action, api_func, services):
+        self.api.responses.append({
+            "change": "70",
+            "result": None,
+            "status": "Accepted",
+            "status-code": 202,
+            "type": "async"
+        })
+        change_id = api_func(timeout=0)
+        self.assertEqual(change_id, '70')
+        self.assertEqual(self.api.requests, [
+            ('POST', '/v1/services', None, {'action': action, 'services': services}),
+        ])
+
+    def test_autostart_services(self):
+        self._test_services_action('autostart', self.api.autostart_services, [])
+
+    def test_autostart_services_async(self):
+        self._test_services_action_async('autostart', self.api.autostart_services, [])
+
+    def test_start_services(self):
+        def api_func():
+            return self.api.start_services(['svc'])
+        self._test_services_action('start', api_func, ['svc'])
+
+    def test_start_services_async(self):
+        def api_func(timeout=30):
+            return self.api.start_services(['svc'], timeout=timeout)
+        self._test_services_action_async('start', api_func, ['svc'])
```

Comment:
One thing that we aren't testing here is what happens if things actually time out.
I'm curious how you would plan to address that code path.